pwa/src/components/TestResultsForm.tsx

```diff
@@ -0,0 +1,134 @@
+                <CheckboxItem
+                    id="blood-sugar-test"
+                    data-testid="blood-sugar-test"
+                    onChange={() => setBloodSugarTest(bloodSugarTest ? 0 : 1)}
+                >
+                    Blood Sugar Test Performed
+                </CheckboxItem>
+                {bloodSugarTest ? (
+                    <div
+                        id="blood-sugar-results"
+                        data-testid="blood-sugar-results"
+                    >
+                        <List renderHeader={() => 'Blood Sugar Results'}>
+                            <InputItem
+                                type="number"
+                                id="blood-sugar-results-val"
+                                data-testid="blood-sugar-results-val"
+                                placeholder="mg/dL" // TOOD: unsure if the unit is mg/dL or mmol/L, could leave this out
+                                maxLength={3}
+                            ></InputItem>
+                            <WhiteSpace className="list-whitespace" size="lg" />
+                        </List>
+                    </div>
+                ) : null}
+                <CheckboxItem
+                    id="ua-test"
+                    data-testid="ua-test"
+                    onChange={() => setUA(ua ? 0 : 1)}
+                >
+                    UA (Urine Analysis) Test Performed
+                </CheckboxItem>
+                {ua ? (
+                    <div id="ua-results" data-testid="ua-results">
+                        <List renderHeader={() => 'UA Results'}>
+                            {/* TODO: I am unsure about the range/type of these fields - check DB */}
+                            <InputItem
+                                id="glucose-results-val"
+                                data-testid="glucose-results-val"
+                                placeholder="0"
+                            >
+                                Glucose
+                            </InputItem>
+                            <InputItem
+                                id="nitrites-results-val"
+                                data-testid="nitrites-results-val"
+                                placeholder="0"
+                            >
+                                Nitrites
+                            </InputItem>
+                            <InputItem
+                                id="protein-results-val"
+                                data-testid="protein-results-val"
+                                placeholder="0"
+                            >
+                                Protein
+                            </InputItem>
+                            <InputItem
+                                id="leukocytes-results-val"
+                                data-testid="leukocytes-results-val"
+                                placeholder="0"
+                            >
+                                Leukocytes
```

Comment:
I attempted to determine what typical values for these tests are via Google, I found that it is likely either `+`, `-` or `[0-9]+`. We should check this with the existing database and update accordingly.

Reply:
After consulting the DB, it appears there is currently a mix of data.

tests_urine_glucose_result was large numbers (165 and 109 are some examples)
test_urinte_nitrites_result is a mix ('7.0', 'neg', 'pos', '+', and 'n' were some examples)
tests_urine_protein_result is also a mix ('6.5', '1', '30', 'neg', '+' were all given)
tests_urine_leukocytes_results is also a mix ('1', 'neg', 'pos', '+' all given)

The expected values/units (and corresponding DB cleaning) should be determined via consult with a CHI rep at a later date and updated accordingly. For now, they will allow any text input.

Reply:
👍